### DISCUSSION TITLE
dagre 自定义react组件用于node  宽高不一样的时候  布局会错乱未对齐

### DISCUSSION THREAD

**Discussion by meikang123** [antvis/X6#1838] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/16129828/155251815-92385898-0d87-465e-b9d5-ee3f8c830ab0.png)

// 布局方向
const dir = 'TB' //

上面和下面无法居中对齐

**Comment by NewByVector** [comment, marked as answer]
因为布局计算出来的是节点的中心点坐标，[布局流程](https://github.com/antvis/X6/blob/master/sites/x6-sites/docs/tutorial/advanced/layout.zh.md#%E5%B8%83%E5%B1%80%E6%B5%81%E7%A8%8B)参考这里
